acvm-repo/acir_field/src/generic_ark.rs

```diff
@@ -272,7 +272,10 @@ impl<F: PrimeField> FieldElement<F> {
     }
     pub fn from_hex(hex_str: &str) -> Option<FieldElement<F>> {
         let value = hex_str.strip_prefix("0x").unwrap_or(hex_str);
-        let hex_as_bytes = hex::decode(value).ok()?;
+        // Values of odd length require an additional "0" prefix
+        let sanitized_value =
+            if value.len() % 2 == 0 { value.to_string() } else { format!("0{}", value) };
+        let hex_as_bytes = hex::decode(sanitized_value).ok()?;
         Some(FieldElement::from_be_bytes_reduce(&hex_as_bytes))
```

Comment:
For reference; we are adding a zero to beginning of the string because we are interpretting it in big endian fomat, see `from_be_bytes_reduce`